docs: document support for fine-grained pat (#27346)
renovatebot/renovate: lib/modules/platform/github/readme.md

## Authentication

First, create a
[fine-grained](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token)
or [classic](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic)
PAT ([Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#about-personal-access-tokens)) for the bot account, select `repo` scope.

Let Renovate use your PAT by doing _one_ of the following:

- Set your PAT as a `token` in your `config.js` file

> - Set your PAT as an environment variable `RENOVATE_TOKEN` (generally via `${{ secrets.RENOVATE_TOKEN }}`)

Reviewer: This is not correct, please revert

Author: If you're using a GitHub workflow, it would be, no?

Author: Err, in my case, I'm not using an env var at all, I'm using the `with:`/`token:`:
https://github.com/check-spelling-sandbox/renovatebot-test-wp-calypso/blob/fc66d304732ee4f7c7ab806d4b64427bfcc55b31/.github/workflows/renovate.yml#L92

```yml
      - uses: renovatebot/github-action@v39.0.1
        with:
          configurationFile: renovate.json5
          token: ${{ secrets.RENOVATE_TOKEN }}
```

Reviewer: This is not the documentation repo for the GitHub Action either though

Author: Fair. Will drop

## Authentication

First, create a
[fine-grained](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token)
or [classic](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic)

> PAT ([Personal Access Token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#about-personal-access-tokens)) for the bot account, select `repo` scope.

Reviewer 2: I think this should all be one line, as it's one sentence. But having it all on a single line is hard to read and maintain because of the long links. Would a short list work?

> First, create one of these:
> 
> - a fine-grained token,
> - or a classic token,
> - or a PAT

It looks like the user can choose between three options? The fine-grained token, a classic token, or a PAT? But the text right after proceeds to mention the PAT first.

So I think we need more changes to the docs than just inserting the fine-grained token stuff.

Author: PAT is a general category. Classic tokens and Fine-grained tokens are *both* PATs.

Reviewer 2: Whoops, you're right! PAT is the _category_, and classic tokens and fine-grained tokens are a item in that category... 🙃 

In any case, I'd still like the sentence to be rewritten so it's easier to maintain.

Author: Doc writing isn't my day job. I do much more editing of other people's docs. I'm happy to apply correct changes and can sometimes iterate through things.

I'd be inclined to have a sentence (that mentions PATs) followed by two bullets because I think it's easier for a reader to parse, but I won't write it unless it'd be acceptable.

Reviewer 2: ```suggestion
First, create a [fine-grained](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#creating-a-fine-grained-personal-access-token) _or_ a [classic](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic) PAT.
The PAT must have the `repo` scope.

Read the [GitHub Docs, about Personal Access Tokens](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#about-personal-access-tokens) to learn more about PATs.
```

Let's go with this then. 😄 

I'm trying to split complicated stuff into their own sentences, so you don't have to remember too much in one go.

    If you're self-hosting Renovate on GitHub.com with GitHub Actions in forking mode, and want Renovate to apply labels then you must give the PAT `triage` level rights on `issues`.
    The `triage` level allows the PAT to apply/dismiss existing labels.

> ## Running using a Fine-grained Token

Reviewer 2: ```suggestion
## Running using a fine-grained Token
```

I think we can lowercase this?

Author: I'd be inclined to lowercase `Token` as well if that's desired.

Whether we can/can't is too complicated for me which is why I didn't respond earlier.

Reviewer 2: ```suggestion
## Running using a fine-grained token
```

This is a _new_ heading, so it should be safe to change the capitalization. 😄

### Permissions

> These are the permissions required for a Fine-grained token:

Reviewer 2: ```suggestion
A fine-grained token must have these permissions:
```

Reviewer 2: It's a bit silly, but I can't apply my own suggestions. Can one of the maintainers apply this for me? 😉

| Permission          | Access           | Level                          |
| ------------------- | ---------------- | ------------------------------ |
| `Members`           | `Read-only`      | _Organization_                 |
| `Commit statuses`   | `Read and write` | _Repository_ or _Organization_ |
| `Contents`          | `Read and write` | _Repository_ or _Organization_ |
| `Dependabot alerts` | `Read-only`      | _Repository_ or _Organization_ |
| `Issues`            | `Read and write` | _Repository_ or _Organization_ |
| `Pull requests`     | `Read and write` | _Repository_ or _Organization_ |
| `Workflows`         | `Read and write` | _Repository_ or _Organization_ |

<!-- prettier-ignore -->
!!! tip "Use a bot role account"

>     You should create a bot role account instead of using a human's github account.

Reviewer: ```suggestion
    Consider creating a GitHub App to use instead of using a "real" GitHub user account.
```
